client/src/sagas/authSagas.js

```diff
@@ -0,0 +1,101 @@
+import axios from 'axios'
+import { call, put, takeLatest } from 'redux-saga/effects'
+
+import { setTokenHeader } from '../actions/authActions'
+import history from '../history'
+import {
+  REQUEST_SIGNUP,
+  REQUEST_SIGNIN,
+  AUTH_ERROR,
+  AUTH_SUCCESS,
+  REQUEST_RESET_PASSWORD,
+  REQUEST_RESET_PASSWORD_SUCCESS,
+  REQUEST_RESET_PASSWORD_ERROR,
+  SUCCESS_PASSWORD_RESET_MSG,
+  REQUEST_NEW_PASSWORD,
+  REQUEST_RESET_NEW_PASSWORD_SUCCESS,
+  REQUEST_RESET_NEW_PASSWORD_ERROR
+} from '../actions/types'
+
+const callSignup = data => {
```

Comment:
I would suggest moving all functions which return axios object, to single file. E.g auth related functions(which return axios object) placed in auth-api.js , User related functions placed in user-api.js , etc.

Reply:
Done!